thewalrus/samples.py

```diff
@@ -87,9 +89,104 @@
 # ===============================================================================================
 
 # pylint: disable=too-many-branches
-def generate_hafnian_sample(
-    cov, mean=None, hbar=2, cutoff=6, max_photons=30, approx=False, approx_samples=1e5
-):  # pylint: disable=too-many-branches
+def decompose_cov(cov):
+    r"""Decompose the convariance matrix using the Williamson decomposition method.
+
+    Args:
+        cov (array): a :math:`2N\times 2N` ``np.float64`` covariance matrix
+            representing an :math:`N` mode quantum state. This can be obtained
+            via the ``scovmavxp`` method of the Gaussian backend of Strawberry Fields.
+
+    Return:
+        T (array): Result of S x S.T.
+        sqrtW (array): Result of S x (D-I).
+    """
+    m = cov.shape[0] // 2
+    D, S = williamson(cov)
+    T = S @ S.T
+    DmI = D - np.eye(2 * m)
+    DmI[abs(DmI) < 1e-11] = 0.0  # remove slightly negative values
+    sqrtW = S @ np.sqrt(DmI)
+    return T, sqrtW
+
+
+def mu_to_alpha(mu, hbar=2):
+    r"""Convert displacement into the mean displacement of each mode.
+
+    Args:
+        mu (array): a :math:`2N`` ``np.float64`` vector of means representing the Gaussian
```

Comment:
Note that there should be double ticks for code-styling but only single ticks for `:math:`. Noticed some places where this was wrong, so it might be worth checking through the docstrings to make sure.
```suggestion
        mu (array): a :math:`2N` ``np.float64`` vector of means representing the Gaussian
```